prov/efa/src/rdm/efa_rdm_ep_progress.c

```diff
@@ -180,8 +180,9 @@ void efa_rdm_ep_progress_post_internal_rx_pkts(struct efa_rdm_ep *ep)
 			ep->efa_rx_pkts_to_post = 0;
 		}
 	} else {
-		if (ep->efa_rx_pkts_posted == 0 && ep->efa_rx_pkts_to_post == 0) {
-			/* Both efa_rx_pkts_posted and efa_rx_pkts_to_post equal to 0 means
+		if (ep->efa_rx_pkts_posted == 0 && ep->efa_rx_pkts_to_post == 0 && ep->efa_rx_pkts_held == 0) {
```

Comment:
should this change be in this commit?

Reply:
sure, I can make it a separate commit. There is another change that decrements the efa_rx_pksts_posted counter in the error path which should be a separate commit too.

Reply:
Updated